modules/installation-azure-preparing-diskencryptionsets.adoc

```diff
@@ -0,0 +1,144 @@
+//Module included in the following assemblies:
+//
+// * installing/installing_azure/enabling-disk-encryption-sets-azure.adoc
+
+:_content-type: PROCEDURE
+[id="preparing-disk-encryption-sets"]
+= Preparing an Azure Disk Encryption Set
+The {product-title} installer can use an existing Disk Encryption Set with a user-managed key. To enable this feature, you can create a Disk Encryption Set in Azure and provide the key to the installer.
+
+.Procedure
+
+. Set the following environment variables for the Azure resource group:
++
+[source,terminal]
+----
+$ export RESOURCEGROUP="<resource_group>"
+----
++
+[source,terminal]
+----
+$ export LOCATION="<location>"
+----
++
+where:
+
+`<resource_group>`:: Specifies the name of the Azure resource group where you will create the Disk Encryption Set and encryption key. You should create the Disk Encryption Set in a different resource group than the resource group where you install the cluster to avoid losing access to your keys after the cluster is destroyed.
+`<location>`:: Specifies the Azure location where you will create the resource group.
++
+. Set the following environment variables for the Azure Key Vault and Disk Encryption Set:
++
+[source,terminal]
+----
+$ export KEYVAULT_NAME="<keyvault_name>"
+----
++
+[source,terminal]
+----
+$ export KEYVAULT_KEY_NAME="<keyvault_key_name>"
+----
++
+[source,terminal]
+----
+$ export DISK_ENCRYPTION_SET_NAME="<disk_encryption_set_name>"
+----
++
+. Set the environment variable for the ID of your Azure Service Principal:
++
+[source,terminal]
+----
+$ export CLUSTER_SP_ID="<service_principal_id>"
+----
++
+where:
+
+`<service_principal_id>`:: Specifies the ID of the service principal you will use for this installation.
++
+. Enable host-level encryption by registering the feature in Azure:
++
+[source,terminal]
+----
+$ az feature register --namespace "Microsoft.Compute" --name "EncryptionAtHost"
+----
++
+[source,terminal]
+----
+$ az feature show --namespace Microsoft.Compute --name EncryptionAtHost
+----
++
+[source,terminal]
+----
+$ az provider register -n Microsoft.Compute
+----
++
+. Create an Azure Resource Group to hold the disk encryption set and associated resources:
++
+[source,terminal]
+----
+$ az group create --name $RESOURCEGROUP --location $LOCATION
+----
++
+. Create an Azure key vault and encryption key:
++
+[source,terminal]
+----
+$ az keyvault create -n $KEYVAULT_NAME -g $RESOURCEGROUP -l $LOCATION \
+        --enable-purge-protection true --enable-soft-delete true
+----
++
+[source,terminal]
+----
+$ az keyvault key create --vault-name $KEYVAULT_NAME -n $KEYVAULT_KEY_NAME \
+        --protection software
+----
++
+. Create a disk encryption set using the key vault and encryption key:
++
+[source,terminal]
+----
+$ KEYVAULT_ID=$(az keyvault show --name $KEYVAULT_NAME --query "[id]" -o tsv)
+----
++
+[source,terminal]
```

Comment:
I think this is a lot of source blocks for one step. Consider breaking this up into multiple steps that are easier to digest.